TypeScript overrides are not working with v4.0.0-alpha.8

## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers |     v4.0.0-alpha.8    |
| material-ui          |     4.10.0    |
| TypeScript           |    3.9.3     |
| React                |     16.13.1    |
| Browser              |    All     |
| Peer library         |    date-fns     |

## Steps to reproduce

Following the [TypeScript example](https://next.material-ui-pickers.dev/guides/css-overrides#typescript) for overriding CSS causes a type error in `@material-ui/core`.

I have the following in my TypeScript declaration file:

```typescript
import { MuiPickersComponentsToClassName } from '@material-ui/pickers/typings/overrides';

declare module '@material-ui/core/styles/overrides' {
  export interface ComponentNameToClassKey extends MuiPickersComponentsToClassName {}
}
```

> Side note: The example also references the `src` path, which is also doesn't work.   Removing `src` seems to make that import work.

## Expected behavior

`tsc` should pass.

## Actual behavior

When I run `tsc` on my project, I get the following error:

```
$ tsc -p tsconfig.json
../../../node_modules/@material-ui/core/styles/overrides.d.ts:100:64 - error TS2344: Type 'ComponentNameToClassKey[Name]' does not satisfy the constraint 'string'.
  Type '"fixed" | "required" | "default" | "square" | "active" | "disabled" | "track" | "error" | "progress" | "message" | "middle" | "media" | "hidden" | "body" | "button" | "caption" | ... 486 more ... | undefined' is not assignable to type 'string'.
    Type 'undefined' is not assignable to type 'string'.

100   [Name in keyof ComponentNameToClassKey]?: Partial<StyleRules<ComponentNameToClassKey[Name]>>;
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

If I remove the picker overrides, `tsc` passes.

--- a/lib/src/typings/overrides.ts
+++ b/lib/src/typings/overrides.ts
@@ -34,35 +34,35 @@ type Classes<T> = T extends string
   : never;
 
 export interface MuiPickersComponentsToClassName {
-  MuiPickersDay?: Classes<typeof DayStyles>;
-  MuiPickerDTHeader?: Classes<typeof DTHeaderStyles>;
-  MuiPickerDTTabs?: Classes<typeof DTTabsStyles>;
-  MuiPickersCalendar?: Classes<typeof CalendarStyles>;
-  MuiPickersCalendarView?: Classes<typeof CalendarViewStyles>;
-  MuiPickersCalendarHeader?: Classes<typeof CalendarHeaderStyles>;
-  MuiPickersSlideTransition?: Classes<typeof SlideTransitionStyles>;
-  MuiPickersYearSelection?: Classes<typeof MuiPickersYearSelectionStyles>;
-  MuiPickersYear?: Classes<typeof MuiPickersYearStyles>;
-  MuiPickersMonthSelection?: Classes<typeof MuiPickersMonthSelectionStyles>;
-  MuiPickersMonth?: Classes<typeof MuiPickersMonthStyles>;
-  MuiPickersTimePickerToolbar?: Classes<typeof TimePickerToolbarStyles>;
-  MuiPickersClock?: Classes<typeof ClockStyles>;
-  MuiPickersClockNumber?: Classes<typeof ClockNumberStyles>;
-  MuiPickersClockPointer?: Classes<typeof ClockPointerStyles>;
-  MuiPickersToolbar?: Classes<typeof PickerToolbarStyles>;
-  MuiPickersToolbarButton?: Classes<typeof ToolbarButtonStyles>;
-  MuiPickersToolbarText?: Classes<typeof MuiPickersToolbarTextStyles>;
-  MuiPickersDatePickerRoot?: Classes<typeof DatePickerRootStyles>;
-  MuiPickerDTToolbar?: Classes<typeof MuiPickerDTToolbarStyles>;
-  MuiBasePickerStyles?: Classes<typeof MuiBasePickerStyles>;
+  MuiPickersDay: Classes<typeof DayStyles>;
+  MuiPickerDTHeader: Classes<typeof DTHeaderStyles>;
+  MuiPickerDTTabs: Classes<typeof DTTabsStyles>;
+  MuiPickersCalendar: Classes<typeof CalendarStyles>;
+  MuiPickersCalendarView: Classes<typeof CalendarViewStyles>;
+  MuiPickersCalendarHeader: Classes<typeof CalendarHeaderStyles>;
+  MuiPickersSlideTransition: Classes<typeof SlideTransitionStyles>;
+  MuiPickersYearSelection: Classes<typeof MuiPickersYearSelectionStyles>;
+  MuiPickersYear: Classes<typeof MuiPickersYearStyles>;
+  MuiPickersMonthSelection: Classes<typeof MuiPickersMonthSelectionStyles>;
+  MuiPickersMonth: Classes<typeof MuiPickersMonthStyles>;
+  MuiPickersTimePickerToolbar: Classes<typeof TimePickerToolbarStyles>;
+  MuiPickersClock: Classes<typeof ClockStyles>;
+  MuiPickersClockNumber: Classes<typeof ClockNumberStyles>;
+  MuiPickersClockPointer: Classes<typeof ClockPointerStyles>;
+  MuiPickersToolbar: Classes<typeof PickerToolbarStyles>;
+  MuiPickersToolbarButton: Classes<typeof ToolbarButtonStyles>;
+  MuiPickersToolbarText: Classes<typeof MuiPickersToolbarTextStyles>;
+  MuiPickersDatePickerRoot: Classes<typeof DatePickerRootStyles>;
+  MuiPickerDTToolbar: Classes<typeof MuiPickerDTToolbarStyles>;
+  MuiBasePickerStyles: Classes<typeof MuiBasePickerStyles>;
   // consider using inline import type notation
-  MuiPickersDesktopDateRangeCalendar?: Classes<
+  MuiPickersDesktopDateRangeCalendar: Classes<
     typeof import('../DateRangePicker/DateRangePickerViewDesktop').useStyles
   >;
-  MuiPickersArrowSwitcher?: Classes<typeof import('../_shared/ArrowSwitcher').useStyles>;
-  MuiPickersDateRangePickerInput?: Classes<
+  MuiPickersArrowSwitcher: Classes<typeof import('../_shared/ArrowSwitcher').useStyles>;
+  MuiPickersDateRangePickerInput: Classes<
     typeof import('../DateRangePicker/DateRangePickerInput').useStyles
   >;
-  MuiPickersModalDialog?: Classes<typeof import('../_shared/PickerModalDialog').useStyles>;
-  MuiPickersCalendarSkeleton?: Classes<typeof import('../CalendarSkeleton').useStyles>;
+  MuiPickersModalDialog: Classes<typeof import('../_shared/PickerModalDialog').useStyles>;
+  MuiPickersCalendarSkeleton: Classes<typeof import('../CalendarSkeleton').useStyles>;
 }

